### ISSUE TITLE
NVDA reports group elements in browsers although this is disabled in document formating settings

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#10635]
### Steps to reproduce:
1. Open a website with group elements
2. Navigate with NVDA in browse mode or focus mode through the website
3. Go to document formating settings and disable "groups"
4. Navigate again through the website

### Actual behavior:
NVDA still reports groups

### Expected behavior:
NVDA should not report groups

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2019.3 Beta 1

#### Windows version:
Windows 10 1909 Update

#### Name and version of other software in use when reproducing the issue:
Firefox 71, Chrome 79

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

**Comment by Adriani90** [comment]
cc: @MarcoZehe, @jcsteh

**Comment by leonardder** [comment]
It looks like I introduced this regression. Unfortunately, I do not have time to look into this just yet.

**Comment by feerrenrut** [comment]
@leonardder could you comment on which PR introduced the regression?

**Comment by leonardder** [comment]
That must have been #10462 or #10499 

I probably missed a check for the formatting config options.

**Comment by leonardder** [comment]
Actually, I just tried to reproduce this with firefox 72 and edge Chromium 79, but couldn't in browse mode. @Adriani90 could you please give an example of a website where you experience this?

Note that in focus mode, groupings are always reported, just as links are, so this is expected behavior.

**Comment by Adriani90** [comment]
It seems I have forgotten to mention it or the behavior has changed.
This is only reproducible when you assign a quick navigation key to groupings and navigate in browse mode by pressing that key. In my case p and shift+p.
Note that this issue does not occur for other containers like lists, tables etc.

**Comment by leonardder** [comment]
I think this is because of #3321. I'm inclined to leave this as is.

**Comment by Cake101** [comment]
Groups are reported when I navigate by heading on some websites, even though that NVDA setting is turned off. After examining the code, it seems there is no way to differentiate a role="group" from a fieldset. Could someone please verify if this is the case?

**Comment by SaschaCowley** [comment]
@Adriani90, @LeonarddeR, @Cake101 could you provide a website and/or minimal example that we can use to reproduce this issue?

**Comment by LeonarddeR** [comment]
You can reproduce this on this particular github issue page. The new comment area is in a grouping. Still reproducible with Firefox.

**Comment by Adriani90** [comment]
Yes, the new comment area is for example in a grouping. When I disable reporting of grouping in document formating settings,
1. NVDA still reports "new comment grouping" when I jump from outside the new comment area to any control within the new comment area by quick nav key, 
or
2. When using tab in focus mode.

I would expect to suppress the announcement of grouping in any case when this is disabled in document formating settings.
In case 2 (focus mode) I think this is out of scope for now, because all other elements are impacted in focus mode as well. E.g. turning off reporting of links works only for browse mode, but not for focus mode. Maybe a solution should be designed globally, so it works both for browse and focus mode in general.

**Comment by Cake101** [comment]
If you go [shopping on Walmart](https://www.walmart.com/ip/Fresh-Whole-Carrots-1-lb-Bag/44391515), under the h2s "Similar items you might like," "More items to consider," and "Customers also considered" are lists of suggested products. Navigating by heading through these products reads out the groups, which is just a bit verbose.
